waiter/src/waiter/handler.clj

```diff
@@ -402,40 +390,24 @@
+              (let [goal "deleted"
+                    sleep-duration-ms 100
+                    delete-result (:result result)
                     response-status (case delete-result
                                       :deleted http-200-ok
                                       :no-such-service-exists http-404-not-found
                                       http-400-bad-request)
-                    router-id->response-chan (when (and (= http-200-ok response-status)
-                                                        (pos? timeout))
-                                               (assoc
-                                                 (make-inter-router-requests-fn (str "apps/" service-id "/await-deletion")
-                                                                                :method :get
-                                                                                :config {:query-string (str "timeout=" timeout)})
-                                                 router-id (async/go
-                                                             {:body (async/go
-                                                                      (json/write-str
-                                                                        {:exists? (async/<! (await-service-deletion-locally
-                                                                                              fallback-state-atom service-id timeout 100))}))})))
-                    router-id->exists? (loop [result {}
-                                              [[router-id response-chan] & remaining] (seq router-id->response-chan)]
-                                         (if (and router-id response-chan)
-                                           (recur
-                                             (assoc result router-id (some-> response-chan
-                                                                          async/<!
-                                                                          :body
-                                                                          async/<!
-                                                                          utils/try-parse-json
-                                                                          (get "exists?")))
-                                             remaining)
-                                           result))
+                    should-poll? (and (= http-200-ok response-status)
+                                      (pos? timeout))
+                    routers-agree (when should-poll?
+                                    (every?
+                                      true?
+                                      (vals
+                                        (<?
+                                          (descriptor/await-service-goal-fallback-state fallback-state-atom make-inter-router-requests-fn router-id service-id timeout sleep-duration-ms goal)))))
```

Comment:
If ` await-service-goal-fallback-state` can't rthow an exception, then we should avoid `fa/<?`

Reply:
see above